flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/JarRunHandler.java

```diff
@@ -103,6 +103,11 @@ public JarRunHandler(
                         executor)
                 .handle(
                         (jobIds, throwable) -> {
+                            try {
+                                program.deleteExtractedLibraries();
```

Comment:
Isn't `JarListHandler` also susceptible to the problem?

Reply:
I am also wondering whether the `ApplicationClusterEntryPoint` really cleans up a `PackagedProgram` after using. Maybe this is something to investigate as a follow up.

Reply:
Yes, the JarListHandler has the same problem. I did not know it actually creates a `PackagedProgram`.

As for the `ApplicationClusterEntryPoint`, the files _should_ be cleaned up when the entry point shuts down because we call `File#deleteOnExit` in `PackagedProgram#createTempFile`.

Reply:
Now whether they are actually deleted depends a bit on when the JVM executes the delete calls. If the ClassLoader is still around at that time, then the deletion may in fact fail.

Explicitly closing the ClassLoader in the PackagedProgram (which unfortunately leaks into the JobGraph...) is a separate issue (see FLINK-9844).

Reply:
Alrighty. Thanks for the update.